ops/cli.py

```diff
@@ -1,351 +1,343 @@
+            info(f"    Available instances in {self.project}:", plain=True)
+            info(shell(f"gcloud --project {self.project} sql instances list", quiet=True), plain=True)
+            issues += 1 if self.force else exit(1)
+        if self.__cluster_exists():
+            success(f"GKE cluster {self.cluster_name} exists")
+        else:
+            error(f"GKE cluster {self.cluster_name} not found")
+            info(f"    Available clusters in {self.project}:", plain=True)
+            info(shell(f"gcloud --project {self.project} container clusters list", quiet=True), plain=True)
+            issues += 1 if self.force else exit(1)
+        if self.__namespace_exists():
+            success(f"Namespace {self.cluster_namespace} exists in cluster")
+            shell(f"gcloud set-context --current --namespace={self.cluster_namespace}")
+            info("    Namespace now set")
+        else:
+            error(f"Namespace {self.cluster_namespace} not found in cluster")
+            info(f"    Available namespaces in {self.cluster_name}:", plain=True)
+            info(shell("kubectl get namespace", quiet=True), plain=True)
+            issues += 1 if self.force else exit(1)
+        if issues != 0:
+            warn(f"{issues} issues but continuing due to '--force'")
+        else:
+            success("Validated configuration")
+
+    def __cluster_exists(self) -> bool:
+        """Check that the cluster exists."""
+        clusters = json.loads(shell(f"gcloud --project {self.project} --format=json "
+                                    f"container clusters list "
+                                    f'{f"--zone {self.cluster_zone}" if self.cluster_zone else ""}'))
+        return self.cluster_name in [c["name"] for c in clusters]
+
+    def __cloudsql_exists(self) -> bool:
+        """Check that the Cloud SQL instanc exists."""
+        instances = json.loads(shell(f"gcloud --project {self.project} --format=json "
+                                     f"sql instances list"))
+        return self.cloudsql_name in [i["name"] for i in instances]
+
+    def __namespace_exists(self) -> bool:
+        """Configure K8s context and check that the namespace is present."""
+        info(f"=>  Configuring Kubernetes for cluster {self.cluster_name}")
+        shell(f"gcloud --project {self.project} container clusters get-credentials "
+              f"{self.cluster_name} --zone {self.cluster_zone}")
+        ctx = f"gke_{self.project}_{self.cluster_zone}_{self.cluster_name}"
+        shell(f"kubectl config use-context {ctx}")
+        try:
+            namespaces = json.loads(shell("kubectl get namespace -o json", timeout=5))
+            return self.cluster_namespace in [item["metadata"]["name"] for item in namespaces["items"]]
+        except subprocess.TimeoutExpired:
+            error("Namespace query took too long--maybe you need to be on non-split VPN?")
+            return False
+
+
+class InfoCommand(Config):
```

Comment:
Inheritance is the source of all evil in OOP. If you read any modern best practices on it, they'll all say avoid it.

Reply:
We face the exact same issue if we pass the result of `parser.parse_args()` around like we currently do, if we so much as change an argument name we have to make changes everywhere. The benefit here is that every IDE and linter perfectly understands the usages here and will identify issues (or do the refactoring for you) whereas if we just pass an untyped object around we have to do the fix manually.

Reply:
"The IDE helps me" is not a good excuse

Reply:
Just because the IDE helps you deosn't mean you should.
Since this is a weakly typed language you are right - objects are pretty much dictionaries. I'm trying to help you in terms of software engineering principles, however.

Reply:
Resolved in new version. No inheritance exists in the file and the `typing` library has been imported to more strongly type the mapping from command names to composite functions.